StackExchange.Profiling.Wcf/WcfMiniProfilerClientInspector.cs

```diff
@@ -84,53 +78,64 @@ public object BeforeSendRequest(ref Message request, IClientChannel channel)
             {
-                // Check to see if we have a request as part of this message
-                MiniProfilerResultsHeader resultsHeader = null;
-                // ReSharper disable PossibleUnintendedReferenceComparison
-                if (reply.Headers.MessageVersion != MessageVersion.None)
-                // ReSharper restore PossibleUnintendedReferenceComparison
+                var headerIndex = reply.Headers.FindHeader(MiniProfilerResultsHeader.HeaderName, MiniProfilerResultsHeader.HeaderNamespace);
+                if (headerIndex >= 0)
                 {
-                    var headerIndex = reply.Headers.FindHeader(MiniProfilerResultsHeader.HeaderName, MiniProfilerResultsHeader.HeaderNamespace);
-                    if (headerIndex >= 0)
-                        resultsHeader = reply.Headers.GetHeader<MiniProfilerResultsHeader>(headerIndex);
+                    resultsHeader = reply.Headers.GetHeader<MiniProfilerResultsHeader>(headerIndex);
                 }
-                else if (_http || reply.Properties.ContainsKey(HttpResponseMessageProperty.Name))
-                {
-                    _http = true;
-
-                    var property = (HttpResponseMessageProperty)reply.Properties[HttpResponseMessageProperty.Name];
+            }
+            else if (_http || reply.Properties.ContainsKey(HttpResponseMessageProperty.Name))
+            {
+                _http = true;
 
-                    var text = property.Headers[MiniProfilerResultsHeader.HeaderName];
-                    if (!string.IsNullOrEmpty(text))
-                        resultsHeader = MiniProfilerResultsHeader.FromHeaderText(text);
-                }
-                else
-                    throw new InvalidOperationException("MVC Mini Profiler does not support EnvelopeNone unless HTTP is the transport mechanism");
+                var property = (HttpResponseMessageProperty)reply.Properties[HttpResponseMessageProperty.Name];
 
-                if (resultsHeader != null && resultsHeader.ProfilerResults != null)
+                var text = property.Headers[MiniProfilerResultsHeader.HeaderName];
+                if (!string.IsNullOrEmpty(text))
                 {
-                    // Update timings of profiler results
-                    if (profilerStart != null)
-                        resultsHeader.ProfilerResults.Root.UpdateStartMillisecondTimingsToAbsolute(profilerStart.StartTime);
-
-                    profiler.AddProfilerResults(resultsHeader.ProfilerResults);
+                    resultsHeader = MiniProfilerResultsHeader.FromHeaderText(text);
                 }
             }
+            else
+            {
+                throw new InvalidOperationException("MVC Mini Profiler does not support EnvelopeNone unless HTTP is the transport mechanism");
+            }
+
+            if (resultsHeader == null || resultsHeader.ProfilerResults == null)
+            {
+                return;
```

Comment:
Why are you returning here without stopping the profiler? isn't the whole idea to add our own timing of there's no explicit timing in the WCF result?

Reply:
You're right, my bad